editor/src/components/editor/actions/actions.ts

```diff
@@ -2088,36 +2097,71 @@ export const UPDATE_FNS = {
             uiFileKey,
             parentPath,
           )
+
+          const parent = MetadataUtils.findElementByElementPath(editor.jsxMetadata, parentPath)
+          const isParentFlex =
+            parent != null ? MetadataUtils.isFlexLayoutedContainer(parent) : false
+
+          function setPositionAttribute(
+            elementToWrapWith: JSXElement,
+            position: 'absolute' | 'relative',
+          ): JSXElement {
+            return {
+              ...elementToWrapWith,
+              props: forceRight(
+                setJSXValueAtPath(
+                  elementToWrapWith.props,
+                  PP.create(['style', 'position']), // todo make it optional
+                  jsxAttributeValue(position, emptyComments),
+                ),
+              ),
+            }
+          }
+
           const targetSuccess = normalisePathSuccessOrThrowError(underlyingTarget)
 
           const withWrapperViewAddedNoFrame = modifyParseSuccessAtPath(
             targetSuccess.filePath,
             editor,
             (parseSuccess) => {
-              const elementToInsert: JSXElement = defaultTransparentViewElement(newUID)
+              const elementToInsert: JSXElement =
+                action.whatToWrapWith === 'default-empty-View'
+                  ? defaultTransparentViewElement(newUID)
+                  : action.whatToWrapWith.element
+
+              const elementToInsertWithPositionAttribute = isParentFlex
+                ? setPositionAttribute(elementToInsert, 'relative')
+                : setPositionAttribute(elementToInsert, 'absolute')
+
               const utopiaJSXComponents = getUtopiaJSXComponentsFromSuccess(parseSuccess)
               const withTargetAdded: Array<UtopiaJSXComponent> = insertElementAtPath(
                 editor.projectContents,
                 editor.canvas.openFile?.filename ?? null,
                 parentPath,
-                elementToInsert,
+                elementToInsertWithPositionAttribute,
                 utopiaJSXComponents,
                 null,
               )
 
               viewPath = EP.appendToPath(parentPath, newUID)
 
+              const importsToAdd: Imports =
+                action.whatToWrapWith === 'default-empty-View'
+                  ? {
+                      // the default View import
+                      ['utopia-api']: {
```

Comment:
Could use `importDetails` here if you wanted to.

Reply:
fixed